### PR TITLE
Basic automated testing for the image with no frills

### PR DESCRIPTION
**Pull request by jaseemabid**
I think this is best we could do right now. This does some basic sanity testing of the image and should catch 80% of the issues with 20% effort. godog jenkins is not worth cleaning up. Triggering a test Jenkinsfile from somewhere else is going to be more work and this seems like the simplest option I can think of now.

### CODE REVIEW

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-2`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-3`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-8`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-9`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-10`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-8`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-19`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-20`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-21`

File: Jenkinsfile [hunk 1]
```diff
@@ -39,9 +39,33 @@ dockerTemplate {
         }
       }
 
+      stage('Test') {
```
Comment (sthaha) [inline Jenkinsfile]: nitpick: The commit message fails to mention what issue it is trying to address.

File: Jenkinsfile [hunk 2]
```diff
@@ -39,9 +39,33 @@ dockerTemplate {
...
+
```
Comment (sthaha) [inline Jenkinsfile]: When does this get executed?

File: Jenkinsfile [hunk 3]
```diff
@@ -39,9 +39,33 @@ dockerTemplate {
...
+            echo 'Container spawned successfully'
+            sh 'cat /etc/redhat-release'
+            sh 'hostname'
+            sh 'java -version'
+            sh 'mvn -version'
+            sh 'git --version'
+
+            // I've no idea where the canonical source for booster apps are, so
+            // let's clone a fork instead
+            sh 'git clone https://github.com/jaseemabid/vertxapp'
```
Comment (sthaha) [inline Jenkinsfile]: can this be a config? also have you considered --shallow ?

**Comment by sthaha** [comment]
@jaseemabid I do like the approach thou

File: Jenkinsfile (hunk 1 above)
Comment (jaseemabid) [reply Jenkinsfile]: @sthaha 'Basic sanity testing for the maven image'. Let's rely less on luck before shipping to prod :)

File: Jenkinsfile (hunk 2 above)
Comment (jaseemabid) [reply Jenkinsfile]: @sthaha Always. During PR and release.

File: Jenkinsfile (hunk 3 above)
Comment (jaseemabid) [reply Jenkinsfile]: We can just change the url when we find out where to clone from. --shallow wont make a difference on this one because its just one or 2 commits there.

Comment (sthaha) [reply Jenkinsfile]: Is this what you mean - https://github.com/openshiftio-vertx-boosters/vertx-http-booster ? please consider using a const instead of a hard coded value.

File: Jenkinsfile [hunk 4]
```diff
@@ -39,9 +39,33 @@ dockerTemplate {
...
+              sh 'mvn clean -B -e -U package -Dmaven.test.skip=false -P openshift'
+            }
+          }
+        }
+      }
+
       if (utils.isCI()) {
         stage('Notify the PR') {
-          if (!env.CHANGE_ID) {
+          def pr = env.CHANGE_ID
```
Comment (sthaha) [inline Jenkinsfile]: How is this related to the commit?  Also, IMHO, this doesn't improve readability

**Comment by sthaha** [comment]
@jaseemabid if you intend to merge this, you may want to remove the DNM label

File: Jenkinsfile (hunk 4 above)
Comment (jaseemabid) [reply Jenkinsfile]: @sthaha This is a regression I introduced earlier. The next line actually refers to this variable. The previous version would actually fail with undefined variable error.

File: Jenkinsfile (hunk 3 above)
Comment (jaseemabid) [reply Jenkinsfile]: @sthaha They are similar but quite different; see Jenkinsfile for example. I'm updating this anyway since I'm not using the changed files

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-23`

**Comment by fabric8cd** [comment]
Snapshot maven-builder image is available for testing.  `docker pull fabric8/maven-builder:SNAPSHOT-PR-37-24`

**Comment by jaseemabid** [comment]
@rupalibehera @sthaha Test passed on master after this merge, so I think that went good. :tada: :tada:  https://jenkins.cd.test.fabric8.io/job/fabric8io-images/job/maven-builder/job/master/
